### PR TITLE
Bip32

### PR DESCRIPTION
**Pull request by tgalal**
This PR adds Bip32 with the following features:

- Parse Extended extended public keys `xpub...`
- Unhardened derivation using string paths such as `m/1/2/3` or `3/4/5` if not starting at depth `0`.
- Unhardened derivation using string an array of indices such as `[1, 2, 3]` to save string parsing cost
- Unhardened derivation of keys at a certain depth with a given index
- Serialization to string `xpub...`

The implementation is based on the following references:

- https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
- https://github.com/ismailakkila/bip32/blob/master/bip32.py
- https://github.com/bitcoin/bitcoin/blob/master/src/util/bip32.cpp

### CODE REVIEW

File: src/Base58.mo [hunk 1]
```diff
@@ -13,6 +14,131 @@ module {
...
+    255,255,255,255,255,255,255,255, 255,255,255,255,255,255,255,255,
+    255,255,255,255,255,255,255,255, 255,255,255,255,255,255,255,255,
+    255,255,255,255,255,255,255,255, 255,255,255,255,255,255,255,255,
+    255,255,255,255,255,255,255,255, 255,255,255,255,255,255,255,255,
+    255,255,255,255,255,255,255,255, 255,255,255,255,255,255,255,255,
+    255,255,255,255,255,255,255,255, 255,255,255,255,255,255,255,255,
+    255,255,255,255,255,255,255,255,
+  ];
+
+  public func decode(input : Text) : [Nat8] {
```
Comment (THLO) [inline src/Base58.mo:36]: Could you please add some documentation to the functions, in particular the public ones?

Comment (tgalal) [reply src/Base58.mo:37]: Added docs.

File: src/Base58.mo [hunk 2]
```diff
@@ -13,6 +14,135 @@ module {
...
+    255,255,255,255,255,255,255,255,
+  ];
+
+  // Convert the given Base58 input to Base256.
+  public func decode(input : Text) : [Nat8] {
+    let inputIter : Iter.Iter<Char> = input.chars();
+    var current : ?Char = inputIter.next();
+    var spaces : Nat = 0;
+
+    // Skip leading spaces
```
Comment (THLO) [inline src/Base58.mo:42]:
```suggestion
    // Skip leading spaces.
```

**Review by THLO — APPROVED** [review]
Looks good to me! Thanks!

**Pull request merged**